### PR TITLE
fix(HMS-1877): Filter authentications by resource type and auth type

### PR DESCRIPTION
**Pull request by adiabramovitch**

### CODE REVIEW

**Comment by lzap** [comment]
Don’t merge let me take a look.

**Comment by adiabramovitch** [comment]
This is a quick fix, but we might use a different endpoint and do something like this: 
`"/sources/v3.1/authentications/?filter\[resource_type\]=Application&filter\[authtype\]\[starts_with\]=provisioning&filter\[resource_id\]=SOURCE_ID"`

**Comment by lzap** [comment]
So it looks like we tested the filtering on `/authentications` endpoint and it works there, however, it does not work on `/sources/XXX/authentications` according to my testing (on stage).

I would rather try to patch sources than reverting back to the filtering that breaks tests.

**Comment by lzap** [comment]
Disregard, I just realized this endpoint only returns info for particular resource, it is gonna be short. It is fine to filter this client side. Sorry, let’s fix tests then.

File: internal/clients/http/sources/sources_client.go [hunk 1]
```diff
@@ -186,14 +185,23 @@ func (c *sourcesClient) GetAuthentication(ctx context.Context, sourceId string)
...
 
-	if len(*resp.JSON200.Data) != 0 {
-		auth := (*resp.JSON200.Data)[0]
-		authentication := clients.NewAuthenticationFromSourceAuthType(ctx, *auth.Username, string(*auth.Authtype), *auth.ResourceId)
-		return authentication, nil
-	} else {
-		logger.Trace().Msgf("Source does not have provisioning authentications of type application")
-		return nil, clients.MissingProvisioningSources
+	// Filter authentications to include only auth where resource_type == "Application". We do this because
+	// Sources API currently does not provide a good server-side filtering.
```
Comment (lzap) [inline internal/clients/http/sources/sources_client.go:189]: Maybe drop a note that we expect this endpoint to never return more than dozen of entries. Just for the record, its three if you enable launch and image builder via super key:

```
$ curl-stage GET "/sources/v3.1/sources/415043/authentications" | jq
{
  "data": [
    {
      "id": "353870",
      "authtype": "access_key_secret_key",
      "username": "XXX",
      "availability_status": "in_progress",
      "resource_type": "Source",
      "resource_id": "415043"
    },
    {
      "id": "353872",
      "authtype": "cloud-meter-arn",
      "username": "arn:aws:iam::XXX:role/redhat-cloud-meter-role-2b61b90877ea2d4f",
      "availability_status": "in_progress",
      "resource_type": "Application",
      "resource_id": "353442"
    },
    {
      "id": "353871",
      "authtype": "provisioning-arn",
      "username": "arn:aws:iam::XXX:role/redhat-provisioning-role-0dbb45280bca3ab0",
      "availability_status": "in_progress",
      "resource_type": "Application",
      "resource_id": "353441"
    }
  ],
  "meta": {
    "count": 3,
    "limit": 100,
    "offset": 0
  },
  "links": {
    "first": "/api/sources/v3.1/sources/415043/authentications?limit=100&offset=0",
    "last": "/api/sources/v3.1/sources/415043/authentications?limit=100&offset=100"
  }
}
```

File: internal/clients/http/sources/sources_client.go [hunk 2]
```diff
@@ -186,14 +185,23 @@ func (c *sourcesClient) GetAuthentication(ctx context.Context, sourceId string)
...
+	auth, err := filterSourceAuthentications(*resp.JSON200.Data)
+	if err != nil {
+		at := make([]string, 0)
```
Comment (lzap) [inline internal/clients/http/sources/sources_client.go:192]:
```suggestion
		var at []string
```

**Comment by lzap** [comment]
Looks like the error is relevant, statuser log:

```
2:34PM WRN Sources did not return any Provisioning authentication for source(auth types): 6([access_key_secret_key]) account_number=0369233 client=sources hostname=provisioning-backend-statuser-59b84df9b5-z2kl2 org_id=3340851 source_id=6 trace_id=d248c7badf04348414d8be967c873fc3 version=c1f6
2:34PM TRC otel: GetAuthentication duration=2.797781 hostname=provisioning-backend-statuser-59b84df9b5-z2kl2 otel=true span_id=ec1cd95d4b3487ff trace_id=5df0c3c60da3e8d4fd562f215455efb1 version=c1f6
2:34PM WRN Not found error from sources error="application read returned no application type in sources: HTTP client: backend service returned not found (404) or no data" account_number=0369233 hostname=provisioning-backend-statuser-59b84df9b5-z2kl2 org_id=3340851 source_id=6 trace_id=d248c7badf04348414d8be967c873fc3 version=c1f6
```

**Comment by ezr-ondrej** [comment]
The test seems to be broken, lets merge and test on stage :)

**Pull request merged**